src/libstd/fs.rs

```diff
@@ -1814,6 +1814,18 @@ mod tests {
         check!(fs::set_permissions(&out, attr.permissions()));
     }
 
+    #[cfg(windows)]
+    #[test]
+    fn copy_file_preserves_streams() {
```

Comment:
In windows NTFS), every file name maps to multiple *actual* files called streams. By default, reading/writing/deleting operates on the "anonymous" stream but you *can* operate on other streams. Alternate streams are kind of like extended attributes but more flexible.

https://technet.microsoft.com/en-us/sysinternals/bb897440.aspx